nuxeo-common/src/test/java/org/nuxeo/common/utils/TestTextTemplate.java

```diff
@@ -157,4 +157,39 @@ public void testEscapeVariable() {
         assertEquals("{$$CZjbsiX748UF583qkbinsQ==}", templates.processText("{$$CZjbsiX748UF583qkbinsQ==}"));
     }
 
+    @Test
+    public void testFreetemplateFileNameValidity() {
+        final TextTemplate textTemplate = new TextTemplate();
+        textTemplate.setFreemarkerParsingExtensions("nxftl");
+
+        final File testResouceDirectory = new File(getClass().getClassLoader().getResource("test-nxftl").getPath());
+        final File tmpdir = new File(System.getProperty("java.io.tmpdir"));
+
+        try {
+            textTemplate.processDirectory(new File(testResouceDirectory, "tutu.xml.nxftl"), tmpdir);
+            textTemplate.processDirectory(new File(testResouceDirectory, "tutu.nxftl"), tmpdir);
+        } catch (Exception e) {
+            Assert.fail("Unexpected exception: " + e.getMessage());
+        }
+
+        final File failingFile = new File(testResouceDirectory, ".nxftl");
+
+        IOException exceptedException = null;
+        try {
+            textTemplate.processDirectory(failingFile, tmpdir);
+        } catch (IOException expected) {
+            exceptedException = expected;
+        } catch (TemplateException e) {
+
+        }
+        Assert.assertNotNull(exceptedException);
+
+        Assert.assertEquals("Template file : '" + failingFile.getAbsolutePath()
```

Comment:
Could you use `String.format` for readability ?
Also put this check under `catch (IOException)` block and remove `TemplateException` block.